lib/cosmos/interfaces/protocols/stream_protocol.rb

```diff
@@ -167,16 +165,10 @@ def log_discard(length, found)
     end
 
     def reduce_to_single_packet
-      if @data.length <= 0
-        # Need some data
-        return :STOP
-      end
-
```

Comment:
I think it could happen if an empty string was passed into read_data().  I'm going to put it back and then merge.  Thanks for the spec updates!